File: runtimes/kotlin-1.6/src/main/kotlin/io/openruntimes/kotlin/RuntimeRequest.kt
package io.openruntimes.kotlin

data class RuntimeRequest(val url: String, val method: String, val scheme: String, val host: String, val port: Int, val path: String, val query: MutableMap<String, String>, val queryString: String, val headers: MutableMap<String, String>, val body: Any, val bodyString: String) {
}

File: runtimes/swift-5.5/Sources/Runtime/RuntimeRequest.swift
import Foundation

class RuntimeRequest {
    let bodyString: String
    let body: Any
    let headers: [String: String]
    let method: String
    let url: String
    let scheme: String?
    let host: String
    let port: Int?
    let path: String
    let queryString: String?
    let query: [String: String]?

    init(
        bodyString: String,
        body: Any,
        headers: [String: String],
        method: String,
        url: String,
        scheme: String?,
        host: String,
        port: Int?,
        path: String,
        queryString: String?,
        query: [String: String]?
    ) {
        self.bodyString = bodyString
        self.body = body
        self.headers = headers
        self.method = method
        self.url = url
        self.scheme = scheme
        self.host = host
        self.port = port
        self.path = path
        self.queryString = queryString
        self.query = query
    }
}
